database_sql_test.go

```diff
@@ -38,7 +39,7 @@ import (
 //
 // In addition to a *db.SQL type, the function returns a cleanup function that
 // should be called when you're done with the database connection.
-func connectDatabaseSQL(
+func connectDatabaseSQLV4(
```

Comment:
We'll be using this in the public documentation. So let's just bump the version to v5 and otherwise not add the same thing for v4.

Reply:
For testing, let's just make sure we have a test for v4 in non-sample code.

Reply:
In theory, v4 and v5 are identical. Still, I'd be a bit more confident if we ran automated tests against both versions, if only for better coverage.

Reply:
Added both tests.